src/scanpy/_utils/__init__.py

```diff
@@ -781,7 +778,7 @@ def _(X: DaskArray, axis: Literal[0, 1]) -> DaskArray:
 
 @overload
 def axis_sum(
-    X: sparse.spmatrix,
+    X: _CSMatrix,
```

Comment:
one of the few cases where really only matrices are meant, will be gone when switching to fast-array-utils (which will never emit np.matrix instances)